csep/core/poisson_evaluations.py

```diff
@@ -315,7 +325,12 @@ def likelihood_test(gridded_forecast, observed_catalog, num_simulations=1000, se
         _ = observed_catalog.region.magnitudes
     except CSEPCatalogException:
         observed_catalog.region = gridded_forecast.region
-    gridded_catalog_data = observed_catalog.spatial_magnitude_counts()
+
+    if isinstance(gridded_forecast.region, QuadtreeGrid2D): #-----Asim's addition----
+        print('Quadtree grid detected.... ')
+        gridded_catalog_data = gridded_forecast.region.get_spatial_magnitude_counts(observed_catalog)
+    else:
+        gridded_catalog_data = observed_catalog.spatial_magnitude_counts()
```

Comment:
see above. also git does a good job of keep track of who added what to the code base. that's nice because it means we don't have to include comments like the `#----` one above.